client/src/shared/components/Pagination/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import Link from 'next/link';
+import ChevronLeft from '@/src/shared/icons/ChevronLeft';
+import ChevronRight from '@/src/shared/icons/ChevronRight';
+
+interface PaginationProps {
+  maxPages: number;
+  totalPages: number;
+  currentPage: number;
+  onChangePage: (page: number) => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  maxPages,
+  totalPages,
+  currentPage,
+  onChangePage
+}) => {
+  const startIndex = Math.max(currentPage - Math.floor(maxPages / 2), 0);
+  const endIndex = Math.min(startIndex + maxPages, totalPages);
+  const pages = Array.from({ length: endIndex - startIndex }, (_, i) => startIndex + i + 1);
+  const prevPage = currentPage - 1;
+  const nextPage = currentPage + 1;
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === totalPages;
+
+  const handlePageChange = (page: number): void => {
+    if (onChangePage != null) {
+      onChangePage(page);
+    }
+  };
+
+  return (
+    <nav>
+      <ul className="flex">
+        {!isFirstPage && (
+          <li className="mr-1">
+            <Link href={`/demo/components/?page=${prevPage}`}>
```

Comment:
- do not render a link for this. we avoid jumping the page from to another.
- let the handler do what he wants to do. (if he wants to jump the page or not)
- for demonstration purposes okay not to jump from one page to another